src/osd/PrimaryLogPG.cc

```diff
@@ -13631,6 +13641,13 @@ bool PrimaryLogPG::agent_work(int start_max, int agent_flush_quota)
       osd->logger->inc(l_osd_agent_skip);
       continue;
     }
+    if (pool.info.cache_mode == pg_pool_t::CACHEMODE_WRITEBACK &&
+        pool.info.cache_pin_head &&
+        p->is_head() && !obc->obs.oi.is_whiteout()) {
+      dout(20) << __func__ << " skip (head) " << obc->obs.oi.soid << dendl;
+      osd->logger->inc(l_osd_agent_skip);
+      continue;
+    }
```

Comment:
can use FLAG_CACHE_PIN to avoid this?

Reply:
can set this flag on the head objects?

Reply:
yes, but it is a rados api for get/set pin-flag, not for admin only.